### PR TITLE
create and clear vector of vertex index per-muon, not per event.

### PR DESCRIPTION
**Pull request by swagata87**

### CODE REVIEW

**Comment by fwyzard** [comment]
I would suggest to
  - leave the declaration of `std::vector<int> vtxInd;` outside the loop, at line 125
  - add `vtxInd.reserve(vtxMuPair.size());` right before line 149
  - add `vtxInd.clear();` at line 179, as you did

**Comment by fwyzard** [comment]
As a separate optimisation, you could modify the constructor of the ScoutingMuon data format to
  - take `vtxIndx` by value rather than `const &`
  - change `vtxIndx_(vtxIndx)` to `vtxIndx_(std::move(vtxIndx))`

**Comment by Martin-Grunewald** [comment]
Need a 92 PR as well.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Dr15Jones** [comment]
+1
For change to scouting

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
@swagata87 
Remember to make a PR for 92X if you need this for 2017 data taking!

File: HLTrigger/Muon/src/HLTScoutingMuonProducer.cc [hunk 1]
```diff
@@ -145,7 +145,8 @@ void HLTScoutingMuonProducer::produce(edm::StreamID sid, edm::Event & iEvent,
       
       double hcalisopf=-1.0;
       if ( HcalPFClusterIsoMap.isValid()) { hcalisopf = (*HcalPFClusterIsoMap)[muonRef]; }
-      
+ 
+      vtxInd.reserve(vtxMuPair.size());
```
Comment (davidlange6) [inline HLTrigger/Muon/src/HLTScoutingMuonProducer.cc]: hi @swagata87 - please move the vtxInd declaration to here - then the clear/reserve calls are not needed

**Comment by fwyzard** [comment]
I'm reading this on my phone, so I may be mistaken, but - why should the
vector be declared inside the loop ?
That would cause additional memory churn.
Calling clear() does not release the memory, so the same buffer can be
reused.

Anyway, even if the vector of declared inside the loop, the reserve is a
good idea, to avoid possible reallocations.

.A

**Comment by davidlange6** [comment]
+1
i agree with @fwyzard

**Pull request merged**